### ISSUE TITLE
default alarm sound fails to play

### ISSUE THREAD

**Issue by forrestguice** [forrestguice/SuntimesWidget#593]
**Describe the bug**
The default alarm sound fails to play and instead the alarm is silent.

**To Reproduce**
1. Add an alarm. Leave the ringtone setting as the default.
2. Wait for the alarm.. When it eventually triggers there is no sound.
3. Repeat these steps, except change the ringtone to any other value. The alarm sound plays as expected.

**Expected behavior**
The default sound should always play (never silent).

**Version Info:**
 - App Version: v0.14.3

**Additional context**
This bug was introduced in `v0.14.3`. It works correctly in previous versions.

**Comment by forrestguice** [comment]
.. always something. :\ 

Its possible most won't notice this because they make a point of setting the alarm sound. And also possible that someone slept through their alarm because of this.

I wish I had noticed this yesterday before tagging `v0.14.4`, and not really sure how I failed to notice it when testing `v0.14.3` .. in fact I swear it was playing correctly then (but since I'm looking at the buggy line that seems unlikely).

**Comment by Timo-Fi** [comment]
It strongly seems to me that Android 11 (Xiaomi, MIUI 12.5) gracefully refuses to show/play any notifications from Suntimes versions 0.14.x,  but does show the notification text from version 0.13.9 and earlier, however this Android 11 device either refuses to play the defined audio, and normally refuses to accept the ringtone / audio file -setting that I make. It will play the default notification sound in case I leave it untouched (if I set it to something else, and then back to the default, then it no longer plays).  I have done the device settings many times, all the notifications related settings are enabled (less vibration), battery optimization is off and DND overriding is allowed. Having the Suntimes Alarms app either on the background or closed has no effect to the above.  I have uninstalled and installed different Suntimes version many times. ... every new Android OS version feeds us some Googlepoo.  Luckily my main phone is Android 7 device, the Suntimes 0.14.3 works fully OK on it. 

When the default notification sound is defined (in Android 7 setup) to be one of my own file, it will be played by Suntimes (I do not know how it it is with the default tones that come hardwired with the OS/GUI, I never use them, they are pain for the ear).  However, if I set the a test notification to be only one or a couple more minutes in the future, it will never trigger. So, I always set test notifications to be over 10 minutes ahead of time, then they are played for sure.  Battery optimization is off.  But then, another app, a talking clock that I once had installed, told me that "your device does not show time accurately".

**Comment by Timo-Fi** [comment]
My bad, the Android 11 (Xiaomi, MIUI 12.5) device does show the notifications from Suntimes. Yesterday I uninstalled the 0.14.3 and then installed 0.14.4, created some notifications for testing, and they played OK.  Likely I have been too impatient in testing.

**Comment by jackthebean** [comment]
> Its possible most won't notice this because they make a point of setting the alarm sound. And also possible that someone slept through their alarm because of this. :\
> 
> I wish I had noticed this yesterday before tagging `v0.14.4`, and not really sure how I failed to notice it when testing `v0.14.3` .. in fact I swear it was playing correctly then (but since I'm looking at the buggy line that seems unlikely).

Looking forward to having it fixed. Thanks.

**Comment by jackthebean** [comment]
I have set up multiple repeating alarms for Civil Twilight start and 
end, both solstices and both equinoxes.

Sometimes, twice so far in about 5 days, I get a notification appearing 
as in the attached.  The first time I swiped the notification (or 
possibly clicked dismiss) and the notification did disappear.  However 
the alarm also did not ring as I expected.

So why :

do these notifications sometimes appear?

why does dismissing them remove that alarm?

Thanks,

Robert Nock



-------- Forwarded Message --------
Date: 	Tue, 27 Sep 2022 11:52:15 +0000 (UTC)
From: 	Robert Nock ***@***.***>
To: 	'Robert Nock' ***@***.***>



Robert Nock
01600 580807
